UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -371,10 +384,15 @@ void OnDisable()
         /// to another (fighting) brain when it comes into contact with an enemy.
         /// </remarks>
         /// <param name="givenBrain">New brain to subscribe this agent to</param>
-        public void GiveBrain(Brain givenBrain)
+        public void GiveModel(NNModel model, string behaviorName)
         {
-            brain = givenBrain;
-            ResetData();
+            m_BrainFactoryParameters.model = model;
```

Comment:
Should this be idempotent? i.e. if the model and behaviorName are the same as the existing one, should we create a new Brain instance, or should it be a no-op?

Reply:
good point, probably no need to create a new Brain in this case.